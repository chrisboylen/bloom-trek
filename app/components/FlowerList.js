import React from 'react';
import {
  ScrollView, StyleSheet, TouchableOpacity, Text,
} from 'react-native';
import { List, ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';

const FlowerList = ({ flowers, goToFlowerDetails }) => (
  <ScrollView>
    {flowers.length
      ? (
        <List containerStyle={styles.list}>
          {
        flowers.map(flower => (
          <TouchableOpacity
            key={flower.name}
            activeOpacity={0.5}
            onPress={() => goToFlowerDetails(flower.id)}
          >
            <ListItem
              roundAvatar
              avatar={{ uri: flower.flower_img_url }}
              title={flower.name}
              subtitle={flower.scientific_name}
              description={flower.description}
              habitat={flower.habitat}
            />
          </TouchableOpacity>
        ))
      }
        </List>
      )
      : <Text style={styles.message}>There are no blooms this month...</Text>}
  </ScrollView>
);

const styles = StyleSheet.create({
  list: {
    marginTop: 0,
    borderTopWidth: 0,
    borderBottomWidth: 0,
  },
  message: {
    marginTop: '30%',
    textAlign: 'center',
    fontSize: 18,
    fontWeight: 'bold',
  },
});

FlowerList.propTypes = {
  flowers: PropTypes.arrayOf(PropTypes.object),
  goToFlowerDetails: PropTypes.func,
};

export default FlowerList;
