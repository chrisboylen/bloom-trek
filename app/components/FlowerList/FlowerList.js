import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';

const FlowerList = ({ flowers, goToFlowerDetails }) => (
  <ScrollView style={styles.container} automaticallyAdjustContentInsets={false}>
    <List>
      {
        flowers.map(flower => (
          <ListItem
            avatar={{ uri: flower.flower_img_url }}
            key={flower.name}
            title={flower.name}
            subtitle={flower.scientific_name}
            description={flower.description}
            habitat={flower.habitat}
            onPress={() => goToFlowerDetails(flower.id)}
          />
        ))
      }
    </List>
  </ScrollView>
);

const styles = StyleSheet.create({
  container: {
    height: '85%',
    // marginTop: -20,
  },
});

FlowerList.propTypes = {
  flowers: PropTypes.arrayOf(PropTypes.object),
  goToFlowerDetails: PropTypes.func,
};

export default FlowerList;
