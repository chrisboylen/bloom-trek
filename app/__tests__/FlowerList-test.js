import React from 'react';
import { shallow } from 'enzyme';
import FlowerList from '../components/FlowerList/FlowerList';

describe('FlowerList', () => {
  let wrapper;
  let mockFlowers;
  let mockGoToFlowerDetails;

  beforeEach(() => {
    mockGoToFlowerDetails = jest.fn();
    mockFlowers = [{
      id: 71,
      flower_img_url: 'http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg',
      name: 'Butter And Eggs',
      common_name: 'Yellow Toadflax',
      scientific_name: 'Linaria vulgaris',
      description: 'Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).',
      bloom_start: 6,
      bloom_end: 10,
      habitat: 'Open areas around former homesteads.',
    }];
    wrapper = shallow(
      <FlowerList
        goToFlowerDetails={mockGoToFlowerDetails}
        flowers={mockFlowers}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke goToFlowerDetails onPress', () => {
    wrapper.find('TouchableOpacity').props().onPress();

    expect(mockGoToFlowerDetails).toHaveBeenCalled();
  });
});
