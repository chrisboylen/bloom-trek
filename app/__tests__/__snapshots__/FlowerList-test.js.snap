// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerList should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlowerList
    flowers={
      Array [
        Object {
          "bloom_end": 10,
          "bloom_start": 6,
          "common_name": "Yellow Toadflax",
          "description": "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
          "flower_img_url": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
          "habitat": "Open areas around former homesteads.",
          "id": 71,
          "name": "Butter And Eggs",
          "scientific_name": "Linaria vulgaris",
        },
      ]
    }
    goToFlowerDetails={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <List
        containerStyle={
          Object {
            "borderBottomWidth": 0,
            "borderTopWidth": 0,
            "marginTop": 0,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={[Function]}
        >
          <ListItem
            avatar={
              Object {
                "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
              }
            }
            chevronColor="#bdc6cf"
            description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
            disabled={false}
            habitat="Open areas around former homesteads."
            hideChevron={false}
            leftIconUnderlayColor="white"
            rightIcon={
              Object {
                "name": "chevron-right",
              }
            }
            rightTitleNumberOfLines={1}
            roundAvatar={true}
            subtitle="Linaria vulgaris"
            subtitleNumberOfLines={1}
            switchButton={false}
            textInputEditable={true}
            title="Butter And Eggs"
            titleNumberOfLines={1}
            underlayColor="white"
          />
        </TouchableOpacity>
      </List>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={[Function]}
          >
            <ListItem
              avatar={
                Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                }
              }
              chevronColor="#bdc6cf"
              description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
              disabled={false}
              habitat="Open areas around former homesteads."
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={true}
              subtitle="Linaria vulgaris"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Butter And Eggs"
              titleNumberOfLines={1}
              underlayColor="white"
            />
          </TouchableOpacity>,
        ],
        "containerStyle": Object {
          "borderBottomWidth": 0,
          "borderTopWidth": 0,
          "marginTop": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Butter And Eggs",
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.5,
            "children": <ListItem
              avatar={
                Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                }
              }
              chevronColor="#bdc6cf"
              description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
              disabled={false}
              habitat="Open areas around former homesteads."
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={true}
              subtitle="Linaria vulgaris"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Butter And Eggs"
              titleNumberOfLines={1}
              underlayColor="white"
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "avatar": Object {
                "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
              },
              "chevronColor": "#bdc6cf",
              "description": "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
              "disabled": false,
              "habitat": "Open areas around former homesteads.",
              "hideChevron": false,
              "leftIconUnderlayColor": "white",
              "rightIcon": Object {
                "name": "chevron-right",
              },
              "rightTitleNumberOfLines": 1,
              "roundAvatar": true,
              "subtitle": "Linaria vulgaris",
              "subtitleNumberOfLines": 1,
              "switchButton": false,
              "textInputEditable": true,
              "title": "Butter And Eggs",
              "titleNumberOfLines": 1,
              "underlayColor": "white",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <List
          containerStyle={
            Object {
              "borderBottomWidth": 0,
              "borderTopWidth": 0,
              "marginTop": 0,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={[Function]}
          >
            <ListItem
              avatar={
                Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                }
              }
              chevronColor="#bdc6cf"
              description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
              disabled={false}
              habitat="Open areas around former homesteads."
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={true}
              subtitle="Linaria vulgaris"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Butter And Eggs"
              titleNumberOfLines={1}
              underlayColor="white"
            />
          </TouchableOpacity>
        </List>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.5}
              onPress={[Function]}
            >
              <ListItem
                avatar={
                  Object {
                    "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                  }
                }
                chevronColor="#bdc6cf"
                description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
                disabled={false}
                habitat="Open areas around former homesteads."
                hideChevron={false}
                leftIconUnderlayColor="white"
                rightIcon={
                  Object {
                    "name": "chevron-right",
                  }
                }
                rightTitleNumberOfLines={1}
                roundAvatar={true}
                subtitle="Linaria vulgaris"
                subtitleNumberOfLines={1}
                switchButton={false}
                textInputEditable={true}
                title="Butter And Eggs"
                titleNumberOfLines={1}
                underlayColor="white"
              />
            </TouchableOpacity>,
          ],
          "containerStyle": Object {
            "borderBottomWidth": 0,
            "borderTopWidth": 0,
            "marginTop": 0,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Butter And Eggs",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.5,
              "children": <ListItem
                avatar={
                  Object {
                    "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                  }
                }
                chevronColor="#bdc6cf"
                description="Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering)."
                disabled={false}
                habitat="Open areas around former homesteads."
                hideChevron={false}
                leftIconUnderlayColor="white"
                rightIcon={
                  Object {
                    "name": "chevron-right",
                  }
                }
                rightTitleNumberOfLines={1}
                roundAvatar={true}
                subtitle="Linaria vulgaris"
                subtitleNumberOfLines={1}
                switchButton={false}
                textInputEditable={true}
                title="Butter And Eggs"
                titleNumberOfLines={1}
                underlayColor="white"
              />,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "avatar": Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                },
                "chevronColor": "#bdc6cf",
                "description": "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
                "disabled": false,
                "habitat": "Open areas around former homesteads.",
                "hideChevron": false,
                "leftIconUnderlayColor": "white",
                "rightIcon": Object {
                  "name": "chevron-right",
                },
                "rightTitleNumberOfLines": 1,
                "roundAvatar": true,
                "subtitle": "Linaria vulgaris",
                "subtitleNumberOfLines": 1,
                "switchButton": false,
                "textInputEditable": true,
                "title": "Butter And Eggs",
                "titleNumberOfLines": 1,
                "underlayColor": "white",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
